### PR TITLE
feat: UI changes for Authz caching

### PR DESCRIPTION
**Pull request by krithika369**
This MR mainly adds a small UI change to warn the users that, when _updating_ permissions for a project, changes may take up to `X` minutes to take effect in the individual components. Since the individual components may set any value for [KeyExpirySeconds](https://github.com/caraml-dev/mlp/blob/main/api/pkg/authz/enforcer/enforcer.go#L47), a new `MaxKeyExpirySeconds` has been introduced in `api/pkg/authz/enforcer/enforcer.go` which is hard-coded to 10 minutes. This means that, components will not be allowed to set key expiry > this value (this will be the `X`).

### Illustration
Screenshot 2023-06-11 at 7 38 53 PM

### Changes
* `api/cmd/main.go` - Supply UI env var `REACT_APP_MAX_AUTHZ_CACHE_EXPIRY_MINUTES`. When doing this, whether Authz and caching are enabled on the MLP app does not matter because they could be turned off on MLP but turned on in Turing, Merlin, etc.
* `api/pkg/authz/enforcer/enforcer.go` - Add `MaxKeyExpirySeconds` and a check to ensure that, when initialising the enforcer, the configured `KeyExpirySeconds` cannot be larger.
* `ui/packages/app/src/config.js` - Add `MAX_AUTHZ_CACHE_EXPIRY_MINUTES` config
* `ui/packages/app/src/project_setting/user_role/SubmitUserRoleForm.js` - Add info in the edit user permissions form

### CODE REVIEW

File: api/config/config.go [hunk 1]
```diff
@@ -190,9 +190,9 @@ var defaultConfig = &Config{
 	Environment: "dev",
 	Port:        8080,
 
-	Streams: Streams{},
-	Docs:    Documentations{},
-
+	Streams:      Streams{},
+	Docs:         Documentations{},
+	Applications: []modelsv2.Application{},
```
Comment (krithika369) [inline api/config/config.go:195]: Minor fix - the value of an uninitialised slice is `nil`.  Though this is rare in practice, this was a causing an issue in the local run of the UI, without any apps.

Screenshot 2023-06-11 at 10 59 34 AM

**Review by deadlycoconuts — APPROVED** [review]
The changes LGTM but just to understand the intention of the changes further, `REACT_APP_MAX_AUTHZ_CACHE_EXPIRY_MINUTES` is expected to be 'fixed' as the default value `600 / 60` minutes right? i.e. we aren't expecting anyone to be manually toggling that UI env var, are we?

**Comment by krithika369** [comment]
> The changes LGTM but just to understand the intention of the changes further, `REACT_APP_MAX_AUTHZ_CACHE_EXPIRY_MINUTES` is expected to be 'fixed' as the default value `600 / 60` minutes right? i.e. we aren't expecting anyone to be manually toggling that UI env var, are we?

Correct. However the env var is still useful when the UI is not being served by the API (as with local dev).

**Comment by krithika369** [comment]
Thank for the quick review, @deadlycoconuts ! Will be merging this shortly.

**Pull request merged**